pkg/matcher/cel.go

```diff
@@ -24,28 +24,75 @@ const (
 	changedFilesTags = "files."
 )
 
+func extractBranchExpression(expression string) map[string]string {
+	regexMatch := regexp.MustCompile(`(\w+)\s*==\s*"([^"]+)"`)
```

Comment:
using regexps inside a CEL expression is akin of trying to screenscrap html page, it's generally a bad idea, any chance we can find another way to do this?...

Reply:
We can do it using strings operation
But thing is if user provides input like

* target_branch == "[$%##refs/heads/*]"
* target_branch == "[       refs/heads/*]"
* target_branch == "[refs/heads/*]"

it will be easy to handle using regex

Otherwise what happens is only valid value for target_branch is 
target_branch == "refs/heads/*"

Other than this whatever user provides that will be invalid

Reply:
I rather not have this kind of code in paac, isn't there a better way to do this ? did you check if cel can do this natively or via a custom function

Reply:
Sure
I tried with CEL and removed string operations 

PTAL